internal/services/signalr/signalr_custom_certificate_binding_resource.go

```diff
@@ -0,0 +1,282 @@
+	return "azurerm_signalr_custom_certificate_binding"
+}
+
+func (r CustomCertBindingSignalrResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var signalRCustomCertBinding CustomCertBindingSignalrModel
+			if err := metadata.Decode(&signalRCustomCertBinding); err != nil {
+				return err
+			}
+			client := metadata.Client.SignalR.SignalRClient
+
+			signalRServiceId, err := signalr.ParseSignalRID(metadata.ResourceData.Get("signalr_service_id").(string))
+			if err != nil {
+				return fmt.Errorf("parsing signalr service id error: %+v", err)
+			}
+
+			keyVaultCertificateId, err := keyVaultParse.ParseOptionallyVersionedNestedItemID(metadata.ResourceData.Get("custom_certificate_id").(string))
+			if err != nil {
+				return fmt.Errorf("parsing custom certificate id error: %+v", err)
+			}
+
+			keyVaultUri := keyVaultCertificateId.KeyVaultBaseUrl
+			keyVaultSecretName := keyVaultCertificateId.Name
+
+			id := signalr.NewCustomCertificateID(signalRServiceId.SubscriptionId, signalRServiceId.ResourceGroupName, signalRServiceId.SignalRName, metadata.ResourceData.Get("name").(string))
+
+			existing, err := client.CustomCertificatesGet(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing SignalR service custom cert binding error %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			customCert := signalr.CustomCertificate{
+				Properties: signalr.CustomCertificateProperties{
+					KeyVaultBaseUri:    keyVaultUri,
+					KeyVaultSecretName: keyVaultSecretName,
+				},
+			}
+
+			if certVersion := keyVaultCertificateId.Version; certVersion != "" {
+				if signalRCustomCertBinding.CertificateVersion != "" && certVersion != signalRCustomCertBinding.CertificateVersion {
+					return fmt.Errorf("certificate version in cert id is different from `certificate_version`")
+				}
+				customCert.Properties.KeyVaultSecretVersion = utils.String(certVersion)
+			}
+
+			if _, err := client.CustomCertificatesCreateOrUpdate(ctx, id, customCert); err != nil {
+				return fmt.Errorf("creating signalR custom certificate binding: %s: %+v", id, err)
+			}
+
+			stateConf := &pluginsdk.StateChangeConf{
+				Pending:    []string{string(signalr.ProvisioningStateCreating), string(signalr.ProvisioningStateUpdating), string(signalr.ProvisioningStateFailed)},
+				Target:     []string{string(signalr.ProvisioningStateSucceeded)},
+				Refresh:    signalrCustomCertBindingStateRefreshFunc(ctx, client, id),
+				MinTimeout: 15 * time.Second,
+				Timeout:    30 * time.Minute,
+			}
+
+			if _, err := stateConf.WaitForStateContext(ctx); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %+v", id, err)
+			}
```

Comment:
would it make sense to use something like `client.CustomCertificatesCreateOrUpdateThenPoll` here?